tests/test_callback_flow.py

```diff
@@ -284,3 +284,73 @@ def test_error_first_cleared_after_succesful_login(
 
     error_response = callback_client.get(error_url)
     assert error_response.status_code == 403
+
+
+@pytest.mark.mock_backend_claims(
+    {
+        "email": "nocollision@example.com",
+        "sub": "some_username",
+    }
+)
+@pytest.mark.auth_request(next="/admin/")
+@pytest.mark.oidcconfig(
+    enabled=True,
+    userinfo_claims_source=UserInformationClaimsSources.id_token,
+    make_users_staff=True,
+)
+def test_set_config_class_on_session_after_successful_login(
+    dummy_config: OpenIDConnectConfig,
+    callback_request: HttpRequest,
+    callback_client: Client,
+    mock_auth_backend: MockBackend,
+):
+    callback_url = reverse("oidc_authentication_callback")
+
+    callback_response = callback_client.get(callback_url, {**callback_request.GET})
+
+    assertRedirects(callback_response, "/admin/", fetch_redirect_response=False)
+    user = User.objects.get(email="nocollision@example.com")
+    assert user.username == "some_username"
+
+    assert (
+        callback_client.session["OIDCDB_CONFIG_CLASS"]
+        == "mozilla_django_oidc_db.OpenIDConnectConfig"
+    )
```

Comment:
I feel that this is testing the implementation details too much. I don't actually care about this key being set in the session, I care about being able to refresh my OIDC session, so the test should:

1. Authenticate against a real keycloak (we have the docker-compose for this)
2. Get redirected back to the admin
3. Then, fetch the admin index again, but `unittest.mock.patch` the middleware check if session refresh is needed (make sure it enters the session refresh flow)
4. assert that we are indeed redirected back to keycloak
5. follow all those redirects etc. and validate that we end up in the admin page again

VCR is of course needed to record the network requests here.